File: ruby/raindrops/raindrops.rb
class Raindrops
  INTEGERS = [3, 5, 7]
  VOYELS = INTEGERS.zip(%w(i a o)).to_h

  def self.convert(number)
    result = INTEGERS.each_with_object("") do |index, string|
      string << "Pl#{VOYELS[index]}ng" if number % index == 0
    end
    result << "#{number}" if result.empty?
    result
  end
end

File: python/raindrops/raindrops.py
def raindrops(number):
    result = ""
    if number % 3 == 0:
        result += "Pling"
    if number % 5 == 0:
        result += "Plang"
    if number % 7 == 0:
        result += "Plong"
    if not (number % 3 == 0 or number % 5 == 0 or number % 7 == 0):
        result = f"{number}"
    return result